cps2pc

```diff
@@ -0,0 +1,226 @@
+    'warn:error': '-Werror', 'nowarn:error': '-Wno-error', 'warn:': '-W',
+    'nowarn:': '-Wno-', 'error:': '-Werror=', 'noerror:': '-Wno-error='}
+_compilers_we_prefixes = {'gcc': _W_we_prefixes, 'clang': _W_we_prefixes}
+_supported_compiler_feature_list = [
+    'c99', 'c11', 'c++03', 'c++11', 'c++14', 'c++17', 'gnu']
+_supported_gcc_features = {
+    x: '--std=%s' % x for x in _supported_compiler_feature_list}
+_supported_clang_features = _supported_gcc_features
+_supported_compile_features = {
+    'gcc': _supported_gcc_features, 'clang': _supported_clang_features}
+
+# Link features
+_supported_link_features = {x: {'threads': '-lpthread'}
+                            for x in _supported_kernels}
+
+#------------------------------------------------------------------------------
+def get_link_features(link_features, platform):
+    try:
+        features = []
+        for feature in link_features:
+            features.append(_supported_link_features[platform][feature])
+        return features
+    except KeyError as k:
+        raise Exception('Unknown link feature %s on %s' %
+                        (k.message, platform))
+
+#------------------------------------------------------------------------------
+def get_compile_features(compile_features, compiler):
+    features = []
+
+    # For loop with catch KeyError exception and check with
+    # WError, and W... flags
+    for feature in compile_features:
+        try:
+            features.append(_supported_compile_features[compiler][feature])
+        except KeyError as k:
+            ma = _re_we_features.match(feature)
+            if ma is not None:
+                features.append(
+                    _compilers_we_prefixes[compiler][ma.group(1)] +
+                    ma.group(5))
+                continue
+            raise Exception(
+                'Unknown link feature %s on %s' % (k.message, compiler))
+    return features
+
+#------------------------------------------------------------------------------
+def get_cpp_runtime_vendor(package):
+    if hasattr(package, 'platform') and \
+            hasattr(package.platform, 'cpp_runtime'):
+        runtime = package.platform.cpp_runtime.vendor
+        try:
+            lib = _supported_cpp_runtime_vendors[runtime]
+            return ['-l%s' % lib]
+        except KeyError:
+            raise Exception('Unknown cpp runtime %s' % runtime)
+    return []
+
+#------------------------------------------------------------------------------
+def get_definitions_flags(compile_definitions, compiler):
+    definitions = []
+    for definition in compile_definitions:
+        try:
+            ma = _re_definitions.match(definition)
+            # No need to test result `ma` as regex catches everything.
+            prefix = _compilers_definition_prefixes[compiler][ma.groups()[0]]
+            definitions.append(prefix + ma.groups()[1])
+        except KeyError:
+            # This should never happen as the list of supported compilers
```

Comment:
If it's a bug, I wonder if it's worth bothering to futz with the exception rather than just let the original exception be raised? Up to you, I guess...